lib/iris/fileformats/netcdf.py

```diff
@@ -510,11 +502,10 @@ def _load_cube(engine, cf, cf_var, filename):
     # Create cube with deferred data, but no metadata
     fill_value = getattr(cf_var.cf_data, '_FillValue', None)
 
-    dtype = nan_array_type(dummy_data.dtype)
-    proxy = NetCDFDataProxy(cf_var.shape, dtype,
+    proxy = NetCDFDataProxy(cf_var.shape, dummy_data.dtype,
```

Comment:
@lbdreyer Calculation of the `fill_value` on `line 503` has the default value of `None`.

In v1.12 and v1.13 it is `netCDF4.default_fillvals[cf_var.dtype.str[1:]]` ... I think that this default behaviour should be reinstated.